### DISCUSSION TITLE
Sandboxing?

### DISCUSSION THREAD

**Discussion by Absolucy** [mlua-rs/mlua#75] [Q&A]
How would sandboxing be done with `mlua`? I don't want scripts to be able to access _any_ files, do _any_ program execution, etc.

**Comment by khvzak** [comment]
You can attach to a chunk an environment table with allowed Lua functions/modules.
Also you can load only desired set of Lua standard libs.
Eg:
```rust
    let lua = Lua::new_with(StdLib::NONE, LuaOptions::default())?;

    let globals = lua.globals();
    let env: Vec<(_, Value)> = vec![
        ("print", globals.get("print")?),
    ];

    lua.load(
        r#"
        print("hello")
        print(require) -- prints nil
        "#,
    )
    .set_environment(lua.create_table_from(env)?)?
    .exec()?;
```

**Comment by ildar** [comment]
Though sandboxing should NOT be relied on as a very secure measure per se.
Some more info: http://lua-users.org/wiki/SandBoxes
